onnxruntime/contrib_ops/rocm/bert/skip_layer_norm_tunable_op.h

```diff
@@ -79,34 +79,37 @@ Status SkipLayerNormStaticSelection(const SkipLayerNormParams<T>* params) {
   const int grid_size = params->element_count / params->ld;
   const int block_size = 256;
 
-#define LAUNCH_SKIPLAYERNORM_SMALL_FORWARD(ELEMENTS, TPB, ILP)                              \
-  if (params->ld <= ELEMENTS) {                                                             \
+#define LAUNCH_SKIPLAYERNORM_SMALL_FORWARD(LOW_ELEMENTS, HIGH_ELEMENTS, TPB, ILP)           \
+  if (params->ld > LOW_ELEMENTS && params->ld <= HIGH_ELEMENTS) {                           \
     SkipLayerNormKernelSmall<T, TPB, ILP><<<grid_size, TPB, 0, params->stream>>>(           \
         params->ld, params->input, params->skip, params->beta, params->gamma, params->bias, \
         maybe2half<T>(params->epsilon), params->output, params->skip_input_bias_add_output, \
         hasBias, hasSkipInputBiasAdditionOutput);                                           \
-    return HIP_CALL(hipPeekAtLastError());                                                  \
```

Comment:
As this `return` is removed, then the chain of `if`s will not have early return semantic, and if any if branches have intersection in their predicate, `SkipLayerNormKernelSmall` will be launch multiple times, which is not safe. Maybe wrap them with `do {} while(0)` and do early `break`.

Reply:
I am OK with current change tho...